File: SEP3_Client/Model/Unit/User/Account.cs
using System;
using SEP3_Client.Model.List.UserList;

namespace SEP3_Client.Model.Unit.User
{
    public class Account : BasicInformation
    {
        public string Password { get; set; }
        public FriendSettingList FriendSettingList { get; set; }
        public Account(string id, string userName,string password) : base(id, userName)
        {
            Password = password;
            FriendSettingList = new FriendSettingList();
        }

        public Account()
        {
            
        }
        
        private Account(string id, string userName, Birthday birthday, string password, FriendSettingList friendSettingList,UserState userState) : this(id, userName, password)
        {
            SetBirthday(birthday);
            SetUserState(UserState);
            FriendSettingList = friendSettingList;
        }

        public bool CheckPassword(string password)
        {
            return Password.Equals(password);
        }

        public bool CheckPassword(Account otherAccount)
        {
            return otherAccount.GetId().Equals(GetId())&&otherAccount.CheckPassword(Password);
        }

        public string ChangePassWord(string oldPassword,string newPassword) 
        {
            if (Password.Equals(oldPassword))
            {
                Password = newPassword;
                return null;
            }
            else
            {
                return "Wrong password.";
            }
        }
        
        public string ChangePassWord(Account oldAccount,Account newAccount) {
            if (GetId().Equals(newAccount.GetId())&&CheckPassword(oldAccount))
            {
                return ChangePassWord(oldAccount.Password,newAccount.Password);
            }
            else 
            {
                return "Wrong change.";
            }
        }

        public FriendSettingList GetFriendSettingList() 
        {
            return FriendSettingList;
        }

        public void SetFriendSettingList(FriendSettingList friendSettingList) 
        {
            FriendSettingList = friendSettingList;
        }

        public void Update(Account account)
        {
            Update(account.GetUserName(),account.GetBirthday());
            FriendSettingList = account.GetFriendSettingList();
        }
        
        public Account Copy() 
        {
            return new Account(GetId(),GetUserName(),GetBirthday(),Password,FriendSettingList.Copy(),UserState.Copy());
        }
    }
}

File: SEP3_UserSystem/src/main/java/com/example/SEP3_UserSystem/model/domain/unit/user/Account.java
package com.example.SEP3_UserSystem.model.domain.unit.user;
import com.example.SEP3_UserSystem.model.domain.list.userList.FriendSettingList;

public class Account extends BasicInformation {
    private String Password;
    private FriendSettingList FriendSettingList;

    public Account(String id, String userName,String password) {
        super(id, userName);
        Password = password;
        FriendSettingList = new FriendSettingList();
    }

    private Account(String id, String userName, Birthday birthday, String password, FriendSettingList friendSettingList,UserState userState) {
        this(id, userName, password);
        setBirthday(birthday);
        setUserState(userState);
        FriendSettingList = friendSettingList;
    }

    public boolean checkPassword(String password)
    {
        if (password==null)
        {
            return false;
        }
        else
        {
            return Password.equals(password);
        }
    }

    public boolean checkPassword(Account otherAccount)
    {
        return otherAccount.getId().equals(getId())&&otherAccount.checkPassword(Password);
    }

    public String changePassWord(String oldPassword,String newPassword) {
        if (Password.equals(oldPassword))
        {
            Password = newPassword;
            return null;
        }
        else
        {
            return "Wrong password.";
        }
    }

    public String changePassWord(Account oldAccount,Account newAccount) {
        if (getId().equals(newAccount.getId())&&checkPassword(oldAccount))
        {
            return changePassWord(oldAccount.Password,newAccount.Password);
        }
        else
        {
            return "Wrong change.";
        }
    }

    public FriendSettingList getFriendSettingList() {
        return FriendSettingList;
    }

    public void setFriendSettingList(FriendSettingList friendSettingList) {
        FriendSettingList = friendSettingList;
    }

    public void update(Account account)
    {
        update(account.getUserName(),account.getBirthday());
        FriendSettingList = account.getFriendSettingList();
    }

    public Account copy() {
        return new Account(getId(),getUserName(),getBirthday(),Password,FriendSettingList.copy(),getUserState().copy());
    }

    public Account toClient() {
        return new Account(getId(),getUserName(),getBirthday(),"",FriendSettingList.copy(),getUserState().copy());
    }
}
